### PR TITLE
Implementation of DFP Service

### PR DESCRIPTION
**Pull request by unasra**

### CODE REVIEW

File: internal/client.go [hunk 1]
```diff
@@ -666,8 +666,25 @@ func (e GenericOpenAPIError) Model() interface{} {
 	return e.model
 }
 
+// WithError ,WithBody and WithModel functions for generic OpenAPIError
```
Comment (mathewab) [inline internal/client.go]: why do you need this functions? are the existing constructors not enough?

File: client/configuration.go [hunk 2]
```diff
@@ -89,7 +89,7 @@ func (c Configuration) internal(basePath string) (*internal.Configuration, error
 	ic := &internal.Configuration{
 		DefaultHeader:    defaultHeaders,
 		UserAgent:        userAgent,
-		Debug:            false,
```
Comment (mathewab) [inline client/configuration.go]: debug is false by default

File: internal/client.go (hunk 1 above)
Comment (unasra) [reply internal/client.go]: resolved

File: client/configuration.go (hunk 2 above)
Comment (unasra) [reply client/configuration.go]: resolved

**Review by mathewab — APPROVED** [review]

**Pull request merged**